### PR TITLE
Change promote branch name from <group>-<app> to env/<group>/<app>.

### LINKED ISSUE #135: Newer PRs replaces older PRs
Consider a `gitops-promotion.yaml`
```
environments:
  - name: dev
    auto: true
  - name: prod
    auto: false
```
When `new` is executed, it will result in a PR#1 that will be auto-merged and applied to `dev`, but the PR#2 from the following `promote` will hang around to be manually merged to `prod`. While it is open, a second image-push will result in a second `new` which will rebase the branch used by PR#2, effectively replacing that PR with one for `dev`. It dutifully updates the title and desc, in GitHub resulting in an entry on the PR saying "@xks-gitops[bot] changed the title **~Promote apps/ze-app in prod to 12345678~ Promote apps/ze-app in dev to abcdef** 1 hour ago". The consequence is that one app can only have a single change in progress across all environments. This may be reasonable in a highly devops environment, when a team owns the full process from dev to ops, but when changes need reviewing by additional stakeholders, this behavior can be confusing.

### PR DESCRIPTION
**Pull request by bittrance**
This means that there can now be one PR pending per app per environment, where there could previously be only one PR per app. Put differently, a change that has been promoted will no longer be replaced by a later new.

Fixes #135. See for details.

Tasks:
- [x] Decide on configurability of name change
- [x] Test coverage for this change (e2e?)
- [x] Revert `dev` tag

### CODE REVIEW

**Comment by bittrance** [comment]
I don't quite trust this change yet; additional testing will be needed.

**Comment by simongottschlag** [comment]
@bittrance should we enable this behind a flag - disabled by default? Then at a later stage after we have been running it for a while in production, we can enable it by default.

**Comment by bittrance** [comment]
@simongottschlag Indeed, because rolling it out may require manually deleting or renaming branches.

**Comment by NissesSenap** [comment]
We are using tags on all our pipelines so if users don't want to update they don't have to.
As we all know we are using many templates on templates features, by using feature flags we will have to update templates not once but 2 times. 

I think it will create two much administration to put this change behind a feature flag but that is my 2 cents.

File: action.yaml [hunk 1]
```diff
@@ -22,7 +22,7 @@ inputs:
     requires: false
 runs:
   using: docker
-  image: docker://ghcr.io/xenitab/gitops-promotion:v0.1.0
+  image: docker://ghcr.io/xenitab/gitops-promotion:dev
   entrypoint: /usr/local/bin/action-entrypoint.sh
```
Comment (simongottschlag) [inline action.yaml:26]: Is this intentional?

File: tests/e2e_test.go [hunk 2]
```diff
@@ -80,8 +80,6 @@ func TestProviderE2E(t *testing.T) {
 			group := "testgroup"
 			app := "testapp"
 
-			promoteBranchName := fmt.Sprintf("promote/%s-%s", group, app)
```
Comment (simongottschlag) [inline tests/e2e_test.go:83]: Don't we need the promote branch name for dev?

**Review by simongottschlag — APPROVED** [review]
Looking good! 🎉🙌

File: action.yaml (hunk 1 above)
Comment (bittrance) [reply action.yaml:26]: In order to actually test this PR, we need to update the `action.yaml` definition to point to a custom image. It goes something like this: gitops-repo/.github/workflows/new.yaml -> gitops-promotion/action.yaml@branch-name -> ghcr.io/image:custom-tag. The `dev` tag was built manually. Will need to remember not to merge this change.

File: tests/e2e_test.go (hunk 2 above)
Comment (bittrance) [reply tests/e2e_test.go:83]: It seems the e2e test doesn't verify that the `new` command creates the branch? Perhaps it should, though?

Comment (bittrance) [reply tests/e2e_test.go:146]: I was confused, of course. Reworked this a bit, so now we check each step properly.

File: action.yaml (hunk 1 above)
Comment (bittrance) [reply action.yaml]: Tag update removed.

**Comment by bittrance** [comment]
In order to get this PR to pass, I have had to update `gitops-promotion.yaml` in the respective testing orgs to include `prflow: per-env` so as to be able to get working e2e tests.

**Pull request merged**